nsv13/code/modules/overmap/fighters/_fighters.dm

```diff
@@ -1730,24 +1691,25 @@ Utility modules can be either one of these types, just ensure you set its slot t
 	var/obj/item/fighter_component/primary/theirGun = them.loadout.get_slot(HARDPOINT_SLOT_PRIMARY)
 	var/obj/item/fighter_component/primary/theirTorp = them.loadout.get_slot(HARDPOINT_SLOT_SECONDARY)
 	//Next up, try to refill the friendly's guns from whatever we have stored in cargo.
-	for(var/atom/movable/AM in hold.contents)
+	for(var/atom/movable/AM as() in hold.contents)
 		if(theirGun.load(them, AM))
 			continue
 		if(theirTorp.load(them, AM))
 			continue
 
-/obj/structure/overmap/fighter/update_icon()
-	cut_overlays()
-	..()
+/obj/structure/overmap/fighter/proc/update_visuals()
+	if(!canopy) //canopy overlay, should always exist
+		canopy = mutable_appearance(icon = icon, icon_state = "canopy_missing")
+		add_overlay(canopy)
 	var/obj/item/fighter_component/canopy/C = loadout?.get_slot(HARDPOINT_SLOT_CANOPY)
-	if(!C)
-		add_overlay(image(icon = icon, icon_state = "canopy_missing", dir = 1))
+	if(QDELETED(C))
+		canopy.icon_state = "canopy_missing"
 		return
 	if(C.obj_integrity <= 20)
-		add_overlay(image(icon = icon, icon_state = "canopy_breach", dir = 1))
+		canopy.icon_state = "canopy_breach"
 		return
 	if(canopy_open)
-		add_overlay("canopy_open")
+		canopy.icon_state = "canopy_open"
```

Comment:
I haven't used mutable_appearance enough to know why, but the canopy icons aren't updating in game. Also the canopy being closed and undamaged is how the actual fighter is drawn, so the overlay needs to be either removed or invisible for that state.